dvc/path_info.py

```diff
@@ -312,3 +312,33 @@ def __eq__(self, other):
             and self._path == other._path
             and self._extra_parts == other._extra_parts
         )
+
+
+class WebdavURLInfo(HTTPURLInfo):
+    def __init__(self, url):
+        super().__init__(url)
+
+    @cached_property
+    def url(self):
+        return "{}://{}{}{}{}{}".format(
+            self.scheme.replace("webdav", "http"),
```

Comment:
what about https? webdavs?

Reply:
If replace in "webdav**S**" webdav on http will be "http**S**"

Reply:
yep, 🤦 sorry :)

Ok, a few concerns still:

1. Do we need to adjust DEFAULT_PORTS in the URLInfo base class?
2. Redefining `url` like this means loosing information. E.g. when we write in logs we will be getting http instead. In general feels like not a good idea creating a discrepancy between what we have inside class and what we return here.

Feels like we should be doing it on the Remote class level, or there should be some separate method - access url or something?

Reply:
1. Fixed

It seems to me to do a separate method is over head. It will be necessary to rewrite the entire code from _HTTP_ with one fix, and at the same time, there will still be the possibility use in logs the wrong _URL_.

Reply:
Yes, it was merely a random suggestion. The whole approach with overriding `url` feels like a hack and can hit us somewhere down the road. There should be a better way of doing this - e.g. RemoteWebDav class should understand how to deal with those URLs (translate them if needed and pass to HTTPRemote), etc.

Reply:
Actually, makes me think if we should tackle it the other way around by simply adding some flag that will tell HTTP(s) remote to use webdav to access this remote that would otherwise have a normal HTTP(s) URL. Something like
```
dvc remote add myremote https://example.com/path
```
and then
```
dvc remote modify myremote webdav true
```
or
```
dvc remote modify myremote type webdav
```
or something else. Not sure if this makes sense. It kinda goes against current conventions, but also I'm not sure if webdav is unique enough for a separate remote, maybe it is better to handle it as HTTP with some special tweaks. :thinking:

Reply:
Hack - not sure. The same **DEFAULT_PORTS** also change the state of the object without notice. The same S3 is HTTP.
Leaving url unchanged - then the purpose of the **_BasePath** class is not very clear. It seemed to me that he should return the ready-made address format for working with him.

As for - as an extension of http. The more I work with him, the more he is everyday. The same gc. I don’t know how it works, but if only the remote method is needed for it, it’s easy to add and now a big difference from HTTP is already obtained.

Reply:
>  The same DEFAULT_PORTS also change the state of the object without notice

Default ports logic is part of the protocols ... e.g. we want to detect that `example.com:80` is the same as `example.com`. In your case you create an object that provides an inconsistent interface - some methods (and some logic internally) might still rely on `webdav`, while url has `http`. It can be very confusing - you ask and object to print its name - it prints Duck but then behave like Dog.

> The same S3 is HTTP

not sure I got this point, could you clarify, please?

> As for - as an extension of http. The more I work with him, the more he is everyday. The same gc. I don’t know how it works, but if only the remote method is needed for it, it’s easy to add and now a big difference from HTTP is already obtained.

Again, not sure I understood this. Btw, we can jump on Discord - dvc.org/chat and discuss this with me and Ruslan (ivan and ruslan there). Please, feel free to ping me/us.